### PR TITLE
Let Navigators have different error codes

### PR DESCRIPTION
**Pull request by RBT22**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | [#3633](https://github.com/ros-planning/navigation2/issues/3633) |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | gazebo simulation |

---

## Description of contribution in a few bullet points

* Change "Failed to get error code: %s from blackboard" log from ERROR to DEBUG
* Add INFO log on initialization about which errors codes are being used to help debug

## Description of documentation updates required from your changes

* 

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

**Comment by RBT22** [comment]
Is there a reason why we use \n between the error codes, instead of spaces @SteveMacenski?  It looks harder to grep this way.

**Comment by SteveMacenski** [comment]
No, I missed that on my scan through, please replace with spaces

File: nav2_behavior_tree/include/nav2_behavior_tree/bt_action_server_impl.hpp [hunk 1]
```diff
@@ -82,6 +82,15 @@ BtActionServer<ActionT>::BtActionServer(
...
       node->set_parameter(error_code_names_param);
+    } else {
+      error_code_names = value.get<std::vector<std::string>>();
+      std::string error_codes_str;
+      for (const auto & error_code : error_code_names) {
+        error_codes_str += "\n" + error_code;
+      }
+      RCLCPP_INFO_STREAM(
+        logger_, "Error_code parameters were set to: "
+          << error_codes_str);
```
Comment (SteveMacenski) [inline nav2_behavior_tree/include/nav2_behavior_tree/bt_action_server_impl.hpp:93]: Can this not fit on the next line up?

**Review by SteveMacenski — COMMENTED** [review]
Generally LGTM

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
Thanks! Looks great!

**Comment by jwallace42** [comment]
I am kind of confused on how this lets navigators have different error codes? All navigators would still be sharing the same set of error codes.

**Comment by SteveMacenski** [comment]
It doesn't, the PR was not well named but named along with the ticket. This is just to silence warnings mostly

**Comment by RBT22** [comment]
Sorry if the naming was not perfect.

> It appears that all navigators would still be sharing the same set of error codes.

I mean the checklist is the same for all of them, but now there is a supported way to propagate different error codes from different Navigators. It wasn't the case before.

**Pull request merged**